tests/test_notifier.py

```diff
@@ -0,0 +1,116 @@
+# pylint: disable=locally-disabled, missing-class-docstring, missing-function-docstring, redefined-outer-name, too-few-public-methods, missing-module-docstring
+
+import os
+import asyncio
+
+import pytest
+
+from jellyfish import Notifier, RoomApi, PeerOptionsWebRTC
+from jellyfish.events import (ServerMessageRoomCreated, ServerMessageRoomDeleted,
+                              ServerMessagePeerConnected, ServerMessagePeerDisconnected,
+                              ServerMessageMetricsReport)
+
+from tests.support.peer_socket import PeerSocket
+from tests.support.asyncio_utils import assert_events, assert_metrics, cancel
+
+
+HOST = 'jellyfish' if os.getenv('DOCKER_TEST') == 'TRUE' else 'localhost'
+SERVER_ADDRESS = f'{HOST}:5002'
+SERVER_API_TOKEN = 'development'
+
+
+class TestConnectingToServer:
+    @pytest.mark.asyncio
+    async def test_valid_credentials(self):
+        notifier = Notifier(server_address=SERVER_ADDRESS,
+                            server_api_token=SERVER_API_TOKEN)
+
+        notifier_task = asyncio.create_task(notifier.connect())
+        await notifier.wait_ready()
+        # pylint: disable=protected-access
+        assert notifier._websocket.open
+
+        await cancel(notifier_task)
+
+    @pytest.mark.asyncio
+    async def test_invalid_credentials(self):
+        notifier = Notifier(server_address=SERVER_ADDRESS,
+                            server_api_token='server_crappy_token')
```

Comment:
Hilarious wordplay (no sarcasm), but I fear this can potentially be viewed as unprofessional by concerned parties...

Reply:
ok ...